bugzilla/comments.go

```diff
@@ -291,26 +294,39 @@ func (s *CommentStore) bugUpdate(obj interface{}, persisted PersistentCommentSto
 		klog.Errorf("Unable to update bug from informer: %v", err)
 		return
 	}
-	if persisted != nil {
-		persisted.NotifyChanged(bug.Info.ID)
+	if s.persistedStore != nil {
+		s.persistedStore.NotifyChanged(bug.Info.ID)
 	}
 }
 
 func (s *CommentStore) bugDelete(obj interface{}) {
 	var name string
+	var id int
+	var err error
 	switch t := obj.(type) {
 	case cache.DeletedFinalStateUnknown:
 		name = t.Key
+		id, err = strconv.Atoi(t.Key)
+		if err != nil {
+			klog.Errorf("Unable to convert string %q to int: %v", t.Key, err)
+			return
+		}
 	case *Bug:
 		name = t.Name
+		id = t.Info.ID
 	default:
 		return
 	}
 	s.lock.Lock()
 	defer s.lock.Unlock()
-	if err := s.store.Delete(&Bug{ObjectMeta: metav1.ObjectMeta{Name: name}}); err != nil {
+	bug := &Bug{ObjectMeta: metav1.ObjectMeta{Name: name}, Info: BugInfo{ID: id}}
+	if err := s.store.Delete(bug); err != nil {
 		klog.Errorf("Unable to delete bug from informer: %v", err)
 		return
 	}
+	if err := s.persistedStore.DeleteBug(bug); err != nil {
```

Comment:
if we do this folks won’t see recently closed bugs and are likely to create new ones.  Is that good or bad?  If we though it was bad, is there a way to simply transition the bug to be searchable as closed (on disk) but purged after time?

Reply:
most bugs don't close all that fast.....  they go to verified, but that's still considered "open".  so for the current release, at least, they stay open for a long time (i believe they close when we GA?).  for previous releases they at least stay open a couple days (until the z-stream ships).

personally i'm not that bothered by someone opening a new bug if the old one got (apparently incorrectly) closed.  There is some overhead in the new bug getting back to the previous assignee, but ultimately the difference between "reopen a closed bug" and "open a new bug" isn't all that great imho.

but if you want what you describe, we can make a modified version of Sync() that checks the bug state prior to deleting it(in addition to seeing that the bug was sufficiently stale) and only deletes closed bugs.  And then we could run that regularly.

Reply:
> but if you want what you describe, we can make a modified version of Sync() that checks the bug state prior to deleting it(in addition to seeing that the bug was sufficiently stale) and only deletes closed bugs. And then we could run that regularly.

well, strike that.  we can only do that if we include closed bugs in the query.  or if we convert "delete this bug" into "mark this bug as closed"

Reply:
so yes, i guess we could make a "persistedStore.CloseBug()" function which rewrote the bug on disk as being in state Closed.  Which could be a lie (we don't know that the bug is closed, we only know that it disappeared from our search.  It's possible it got moved to an entirely different product or something).

but it sounds like it would mostly get the behavior you want.